core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestRemoveTrivialFilters.java

```diff
@@ -57,10 +57,11 @@ public void testRemovesNullFilter()
     {
         tester().assertThat(new RemoveTrivialFilters())
                 .on(p -> p.filter(
-                        expression("null"),
```

Comment:
Regarding the connection of this PR and https://github.com/trinodb/trino/pull/15558, can you also remove the handling of this case from `RemoveTrivialFilters` and then whoever gets the conflict will solve it, or do you have other plans?

Reply:
> However, I think that the Optimizer rule should not validate the node, but instead just ignore `NullLiteral`.

Yes, a rule doesn't need to validate a node, but it also is free to fail when a node doesn't adhere to a design. For example, a ProjectNode cannot have multiple sources, so a rule can simply fail if one does.

> I think that for now we can skip the validation altogether. That would be consistent with how we handle other sites where boolean type is required

answered at the other comment: https://github.com/trinodb/trino/pull/15744/commits/020b3662c8d3518b1aa8f9fb8cd62ffe03ad1879#r1085170555